src/keboola/utils/date.py

```diff
@@ -0,0 +1,180 @@
+import dateparser
+import math
+import pytz
+from _datetime import timedelta
+from datetime import datetime
+from dateutil.relativedelta import relativedelta
+from typing import Tuple, Generator, Union, Dict, List
+
+date_tuple = Union[Tuple[datetime, datetime], Tuple[str, str]]
+date_gen = Generator[Dict[str, datetime], None, None]
+date_chunk = Union[date_gen, List]
+
+
+def get_date_period_converted(period_from: str, period_to: str, strformat: str = None) -> date_tuple:
+    """
+    Returns given period parameters in datetime format, or next step in back-fill mode
+    along with generated last state for next iteration.
+
+    Args:
+        period_from: YYYY-MM-DD or relative string supported by date parser e.g. 5 days ago
+        period_to: YYYY-MM-DD or relative string supported by date parser e.g. 5 days ago
+        strformat: A python strtime format, in which output dates will be returned. If not specified
+                    function returns dates in datetime.datetime type
+
+    Returns:
+        start_date: datetime, end_date: datetime
+    """
+
+    start_date_form = dateparser.parse(period_from)
+    end_date_form = dateparser.parse(period_to)
+    day_diff = (end_date_form - start_date_form).days
+    if day_diff < 0:
+        raise ValueError("start_date cannot exceed end_date.")
+
+    if strformat is None:
+        return start_date_form, end_date_form
+    else:
+        return start_date_form.strftime(strformat), end_date_form.strftime(strformat)
+
+
+def get_backfill_period(period_from: str, period_to: str, last_state: dict) -> date_tuple:
```

Comment:
I think this can be remocved, I used it maybe once and it's dependent on the internal structure of the last_state dict. What do you think?

Reply:
Got it and removed it.